poly/src/polynomial/univariate/dense.rs

```diff
@@ -509,8 +509,7 @@ impl<'a, F: Field> Sub<&'a SparsePolynomial<F>> for &DensePolynomial<F> {
     #[inline]
     fn sub(self, other: &'a SparsePolynomial<F>) -> DensePolynomial<F> {
         if self.is_zero() {
-            let result = other.clone();
-            (-result).into()
+            (-other.clone()).into()
```

Comment:
I think to emphasize the order of operations, this should be 
```suggestion
            (-(other.clone())).into()
```
Since negation takes its argument by value.